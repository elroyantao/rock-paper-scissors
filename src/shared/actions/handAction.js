export const setPlayerHand = (playerHand) => ({
  type: 'SET_PLAYER_HAND',
  playerHand
})

export const setComputerHand = (computerHand) => ({
  type: 'SET_COMPUTER_HAND',
  computerHand
})

export const clearHand = () => ({
  type: 'CLEAR_HANDS'
})
