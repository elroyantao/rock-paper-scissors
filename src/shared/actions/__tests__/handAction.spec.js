import * as actions from '../handAction'

const snapshot = (action) => expect(action).toMatchSnapshot()

describe('handAction', () => {
  it('setPlayerHand', () => {
    snapshot(actions.setPlayerHand(1))
  })
  it('setComputerHand', () => {
    snapshot(actions.setComputerHand(2))
  })
  it('clearHand', () => {
    snapshot(actions.clearHand())
  })
})
