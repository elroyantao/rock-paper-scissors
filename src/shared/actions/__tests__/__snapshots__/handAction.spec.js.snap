// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handAction clearHand 1`] = `
Object {
  "type": "CLEAR_HANDS",
}
`;

exports[`handAction setComputerHand 1`] = `
Object {
  "computerHand": 2,
  "type": "SET_COMPUTER_HAND",
}
`;

exports[`handAction setPlayerHand 1`] = `
Object {
  "playerHand": 1,
  "type": "SET_PLAYER_HAND",
}
`;
